pkg/deploy/util/util.go

```diff
@@ -104,6 +104,27 @@ func HasChangeTrigger(config *deployapi.DeploymentConfig) bool {
 	return false
 }
 
+// HasImageChangeTrigger returns whether the provided deployment config has
+// an image change trigger or not.
+func HasImageChangeTrigger(config *deployapi.DeploymentConfig) bool {
+	for _, trigger := range config.Spec.Triggers {
+		if trigger.Type == deployapi.DeploymentTriggerOnImageChange {
+			return true
+		}
+	}
+	return false
+}
+
+// GetImageChangeTrigger returns the image change trigger from a deployment config.
+func GetImageChangeTrigger(config *deployapi.DeploymentConfig) deployapi.DeploymentTriggerPolicy {
```

Comment:
A deployment config has multiple triggers, so this is not correct if only the second or third trigger is automatic. Please add a test and update this.